### PR TITLE
xz-git: Add missing provides and conflicts fields.

### PR DESCRIPTION
**Pull request by fracting**
Built and tested on Travis CI: https://travis-ci.org/fracting/xz/jobs/73845661#L3037

### CODE REVIEW

File: xz-git/PKGBUILD [hunk 1]
```diff
@@ -11,6 +11,8 @@ url='http://tukaani.org/xz/'
 license=('GPL' 'LGPL' 'custom')
 depends=('sh' 'libiconv' 'gettext')
 makedepends=('libiconv-devel' 'gettext-devel')
+provides=("${_realname}")
+conflicts=("${_realname}")
```
Comment (Alexpux) [inline xz-git/PKGBUILD]: you need add ````provides```` and ````conflicts```` of every package part in ````package_*```` functions instead header

**Comment by fracting** [comment]
Thanks Alexpux!

I update `provides` and `conflicts` according to your advice, however, I have no idea whether should I use `${pkgver}` in provides/conflicts like libtre-git does, any further advice?

I also fix broken git repository URL, sorry for my mistake, it was not tested since I used --noextract in Travis CI build.

Thanks!

File: xz-git/PKGBUILD [hunk 2]
```diff
@@ -73,6 +77,9 @@ package_liblzma-devel-git() {
   pkgdesc="Liblzma headers and libraries"
   groups=('development')
   depends=("liblzma=${pkgver}" 'libiconv-devel' 'gettext-devel')
```
Comment (Alexpux) [inline xz-git/PKGBUILD]: You need depends on the same package parts. In you case ````depends=("liblzma-git=${pkgver}...````

**Comment by fracting** [comment]
Hi Alexpux, thanks for feedback, I update the patch and tested with https://ci.appveyor.com/project/fracting/msys2-packages/build/1.0.61#L1902

Could you review again?

(BTW, I'm still waiting for feedback on my appveyor proposal, once we integrate an appveyor solution to your repo, we can easily test every pull request in the future.)

**Comment by Alexpux** [comment]
Looks ok, but rebase your changes for current repo state.

**Comment by fracting** [comment]
Rebased, thanks!

**Pull request merged**